### PR TITLE
DEVHUB-595

### PR DESCRIPTION
**Pull request by Dmytro-Shustra-EPAM**
[Staging Link to Branch](https://docs-mongodbcom-staging.corp.mongodb.com/CI/devhub/jordanstapinski/DEVHUB-595-sentiment-score/)

This PR adds article rating and sending some data to the server. Adds contexts for better data flow and has some fixes for Modal. Components have common logic via context, so it's can be a little confused.

### CODE REVIEW

File: src/components/dev-hub/modal.js [hunk 1]
```diff
@@ -112,15 +114,18 @@ export const Modal = ({
...
+    }
     const closeModalOnEnter = e => {
         const enterKey = 13;
         if (e.keyCode === enterKey) {
             deactivateModal();
         }
     };
     const theme = useTheme();
-    const ResponsiveModal = () =>
+    const responsiveModal = () =>
```
Comment (Dmytro-Shustra-EPAM) [inline src/components/dev-hub/modal.js:128]: It should help to stop creating ResponsiveModal for every render.

**Review by jestapinski — COMMENTED** [review]
First pass of comments, will add some more later on but overall working well!

File: src/components/dev-hub/feedback/feedback-final-step.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    justify-content: center;
+    margin-bottom: ${size.medium};
+`;
+
+const FeedbackFinalStep = ({ incrementStep }) => (
+    <>
+        <StyledTitle>We appreciate your feedback.</StyledTitle>
+        <StyledDescription>
+            We'd love to chat with you and answer your
+            questions in our online <Link to="/community/forums/">MongoDB Community</Link>.
```
Comment (jestapinski) [inline src/components/dev-hub/feedback/feedback-final-step.js:37]:
```suggestion
            questions in our online <Link href="https://developer.mongodb.com/community/forums/">MongoDB Community</Link>.
```

Ah I should have mentioned. Although we share a subdomain, this is a separate app so we cannot internal link to it

File: src/components/dev-hub/modal.js (hunk 1 above)
Comment (jestapinski) [reply src/components/dev-hub/modal.js:128]: ah I see, so we kept creating a new modal for each render? I'm assuming this made the callback fire more often than we wanted.

File: src/components/ArticleRatingContext.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    return { ...nextState };
+};
+
+const ArticleRatingProvider = ({ children }) => {
+    const [ratingState, dispatch] = useReducer(ratingReducer, initialState);
+
+    const ratingDispatch = (type, value) => {
+        dispatch({ type: type, value: value });
```
Comment (jestapinski) [inline src/components/ArticleRatingContext.js:61]: Small nitpick, we can get away with the shorthand syntax here

```suggestion
        dispatch({ type, value });
```

**Review by jestapinski — COMMENTED** [review]
Looks really awesome! This has been coming together quite well. I had some smaller comments.

Not sure if you saw my comment on the modal being full height for mobile? It should be full height and width.

This is a bit hard to see, but when the modal is open the page width changes so it no longer spans full width. Is this an issue with the modal component in general?

Screen Shot 2021-04-21 at 5 41 17 PM

File: src/components/dev-hub/feedback/feedback-final-step.js [hunk 4]
```diff
@@ -29,12 +29,14 @@ const StyledButtonContainer = styled('div')`
     margin-bottom: ${size.medium};
 `;
 
+const FORUMS_URL = 'https://developer.mongodb.com/community/forums/';
```
Comment (jestapinski) [inline src/components/dev-hub/feedback/feedback-final-step.js:32]: We can also use this from the `constants.js` file

**Review by jestapinski — APPROVED** [review]
Great! I added some styling for the mobile modal which should help us get more of the expected style from Figma. Take a look and, when you are ready, feel free to merge.

Thanks Dmytro!

**Pull request merged**